BedrockServer.cpp

```diff
@@ -354,12 +361,16 @@ void BedrockServer::sync(SData& args,
             if (nodeState == SQLiteNode::STANDINGDOWN) {
                 continue;
             }
-
             // Now we can pull the next command off the queue and start on it.
             command = syncNodeQueuedCommands.pop();
             SINFO("[performance] Sync thread dequeued command " << command.request.methodLine << ". Sync thread has "
                   << syncNodeQueuedCommands.size() << " queued commands.");
 
+            // Set the function that lets the signal handler know which command caused a problem, in case that happens.
+            SSetSignalHandlerDieFunc([&](){
+                server._syncNode->broadcastImmediate(_generateBlacklistMessage(command));
+            });
```

Comment:
I'm not following the logic here.  So if the worker thread gets a signal it will immediately send the message, but if the sync thread gets a signal it just sets a flag such that the message will be sent later?